common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -409,19 +472,34 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
-                    break;
+                    return DeadlineRunResult.TIMED_OUT;
                 }
             }
 
-            task = pollTask();
+            task = pollTaskFrom(taskQueue);
             if (task == null) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
```

Comment:
if the code stays as is we shouldn't need a temporary variable for `lastExecutionTime`. We could just change this to:

```java
executionTimeCallback.accept(startTime, ScheduledFutureTask.nanoTime());
return DeadlineRunResult.RAN_AT_LEAST_ONCE;
```